apiserver/facades/client/application/updateseries.go

```diff
@@ -136,22 +136,17 @@ func (s stateSeriesValidator) ValidateApplication(application Application, base
 	if err != nil {
 		return errors.Trace(err)
 	}
-	supportedSeries, err := corecharm.ComputedSeries(ch)
+	supportedBases, err := corecharm.ComputedBases(ch)
 	if err != nil {
 		return errors.Trace(err)
 	}
-	if len(supportedSeries) == 0 {
-		supportedSeries = append(supportedSeries, ch.URL().Series)
+	if len(supportedBases) == 0 {
+		err := errors.NewNotSupported(nil, fmt.Sprintf("charm %q does not support any bases. Not valid", ch.Meta().Name))
+		return apiservererrors.ServerError(err)
 	}
```

Comment:
It does not atm. Perhaps it should though?

In reality, if a charm has no series/bases, it will fail long before this, so I don't think it's too important

Reply:
@hmlanigan thoughts?